### PR TITLE
Fixes problem with ValueJoins and arrays

### PR DESCRIPTION
**Pull request by systay**
Array equality is not simple in the JVM. This PR fixes problems when value joins are used on array properties

### CODE REVIEW

File: community/cypher/cypher-compiler-3.1/src/main/scala/org/neo4j/cypher/internal/compiler/v3_1/pipes/ValueHashJoinPipe.scala [hunk 1]
```diff
@@ -49,7 +50,8 @@ case class ValueHashJoinPipe(lhsExpression: Expression, rhsExpression: Expressio
     val result = for {context: ExecutionContext <- rhsIterator
                       joinKey = rhsExpression(context) if joinKey != null}
       yield {
-        val seq = table.getOrElse(joinKey, mutable.MutableList.empty)
+        val equiKey = Equivalent(joinKey)
```
Comment (Mats-SX) [inline community/cypher/cypher-compiler-3.1/src/main/scala/org/neo4j/cypher/internal/compiler/v3_1/pipes/ValueHashJoinPipe.scala:53]: Are value joins an `equivalence` thing? I thought they belonged with `equality`?

**Comment by pontusmelke** [comment]
@systay this has been red a while now

File: community/cypher/cypher-compiler-3.1/src/main/scala/org/neo4j/cypher/internal/compiler/v3_1/pipes/ValueHashJoinPipe.scala (hunk 1 above)
Comment (systay) [reply community/cypher/cypher-compiler-3.1/src/main/scala/org/neo4j/cypher/internal/compiler/v3_1/pipes/ValueHashJoinPipe.scala:53]: Maybe the code base is naming things wrongly - this PR does the same thing that `Equals` does: https://github.com/neo4j/neo4j/blob/3.1/community/cypher/cypher-compiler-3.1/src/main/scala/org/neo4j/cypher/internal/compiler/v3_1/commands/predicates/ComparablePredicate.scala#L71

**Comment by systay** [comment]
@pontusmelke It's green now. A rebase was all it took.

**Pull request merged**